### PR TITLE
(fix) O3-2832 : Implemented validation check for Primary Diagnosis and Secondary Diagnosis in visit-note

### PR DESCRIPTION
**Pull request by senthil-k8s**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
Duplication occurs while selecting the Primary and Secondary Diagnosis in visit-note page.

## Screenshots

Before Changes:
https://www.loom.com/share/89234204c7a74591a6e1ffde949639f7?sid=8c770d7d-c0e6-40dc-9e6a-fa05c06d06c6

After Changes:
https://www.loom.com/share/72557229c50e43a08dcc4606769a0a79?sid=62b7ed9f-e6c0-42fa-a14e-314723e62ae7

## Related Issue
[O3-2832](https://openmrs.atlassian.net/jira/software/c/projects/O3/issues/O3-2832)

### CODE REVIEW

**Review by balaji-sivasakthi — APPROVED** [review]
LGTM 👍🏻

**Comment by senthil-k8s** [comment]
Hey @vasharma05. Thanks for your review. I thought that would be an efficient solution. Now, I'm clear about it. I'll make sure that option is not repeated again.

**Comment by senthil-k8s** [comment]
Hey @vasharma05, Correct me if I am wrong. If the user selected an option in primary diagnosis, shouldn't the selected diagnosis be available in the secondary diagnosis drop-down list?

**Comment by senthil-k8s** [comment]
Hey @vasharma05 , Changes made based on your review

### Screenshot:
https://www.loom.com/share/2a100cd94da9447a8ba752acebb31e7c?sid=fa2308d0-4c7f-45ed-94f8-c80557e57230

File: packages/esm-patient-notes-app/src/notes/visit-notes-form.component.tsx [hunk 1]
```diff
@@ -407,16 +407,26 @@ const VisitNotesForm: React.FC<DefaultWorkspaceProps> = ({ closeWorkspace, patie
...
-                            className={styles.diagnosis}
-                            key={index}
-                            onClick={() => handleAddDiagnosis(diagnosis, 'primaryDiagnosisSearch')}
-                          >
-                            {diagnosis.display}
-                          </li>
-                        ))}
+                        {searchPrimaryResults.map((diagnosis, index) => {
+                          const isPrimaryDiagnosisSelected = selectedPrimaryDiagnoses.some(
+                            (selectedDiagnosis) => diagnosis.display === selectedDiagnosis.display,
```
Comment (vasharma05) [inline packages/esm-patient-notes-app/src/notes/visit-notes-form.component.tsx:412]: Change looks how it should be.
2 small nits:
1. Can we have a function that looks into both primary and secondary diagnoses? The same code is used in 2 places.
2. Please compare the diagnoses not by `display` but by `UUID`.

Thanks!

Comment (senthil-k8s) [reply packages/esm-patient-notes-app/src/notes/visit-notes-form.component.tsx:412]: Sure!

**Review by vasharma05 — APPROVED** [review]
Looks good to me!
Thanks @senthil-k8s!

**Pull request merged**